frontend/src/components/elements/VegaLiteChart/VegaLiteChart.test.tsx

```diff
@@ -46,6 +124,49 @@ describe("VegaLiteChart Element", () => {
     expect(wrapper.find("StyledVegaLiteChartContainer").length).toBe(1)
   })
 
+  const cases = [
+    [baseCase, case1, false, "Case 1: number of columns don't match"],
+    [baseCase, case2, false, "Case 2: same number of rows"],
+    [baseCase, case3, false, "Case 3: previous number of rows greater"],
+    [case4, baseCase, false, "Case 4: no previous rows"],
+    [
+      baseCase,
+      case5,
+      false,
+      "Case 5: light comparison fails, changed value of last column, first row",
+    ],
+    [
+      baseCase,
+      case6,
+      false,
+      "Case 6: light comparison fails, changed value of last column, second to last row",
+    ],
+    [baseCase, case7, true, "Case 7: data is an append"],
+  ]
+
+  cases.forEach(test => {
+    it(`tests for appended data properly - ${test[3]}`, () => {
+      const prevData = test[0]
+      const data = test[1]
+      const expected = test[2]
+      const [prevNumRows, prevNumCols] = tableGetRowsAndCols(
```

Comment:
This can be cleaned up with some destructuring. This helps with just better naming.

```javascript
cases.forEach(([prevData, data, expected, testDescription]) => {
  it(`tests for appended data properly - ${testDescription}`, () => {
    const [prevNumRows, prevNumCols] = tableGetRowsAndCols(prevData.get("data"))
    // ...
  })
})

This allows us to reduce the lines of code in a more descriptive way rather than reference array locations.

Reply:
Really appreciate the review & feedback 😄